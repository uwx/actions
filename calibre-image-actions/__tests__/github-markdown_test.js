const results = {
  optimisedImages: [
    {
      name: 'icon.png',
      path: '__tests__/test-images/icon.png',
      beforeStats: 8914,
      afterStats: 3361,
      percentChange: -62.29526587390622,
      compressionWasSignificant: true
    },
    {
      name: 'roo.jpg',
      path: '__tests__/test-images/roo.jpg',
      beforeStats: 485742,
      afterStats: 468895,
      percentChange: -3.592915258213452,
      compressionWasSignificant: true
    }
  ],
  unoptimisedImages: [
    {
      name: 'optimised-image.png',
      path: '__tests__/test-images/optimised-image.png',
      beforeStats: 3361,
      afterStats: 3361,
      percentChange: 0,
      compressionWasSignificant: false
    },
    {
      name: 'another-optimised-image.png',
      path: '__tests__/test-images/another-optimised-image.png',
      beforeStats: 3361,
      afterStats: 3361,
      percentChange: 0,
      compressionWasSignificant: false
    }
  ],
  metrics: {
    bytesSaved: 5553,
    percentChange: -62.29526587390622
  }
}

const markdown = require('../dist/github-markdown').default

test('writes the markdown with diff report', async () => {
  const markdownResult = await markdown({
    processingResults: results,
    commitSha: 'a033d6f26da7f7856c150e7f1bf217f0f0cfd7e3'
  })

  expect(markdownResult).toMatchSnapshot()
})

test('writes the markdown without diff report', async () => {
  const markdownResult = await markdown({
    processingResults: results
  })

  expect(markdownResult).toMatchSnapshot()
})
