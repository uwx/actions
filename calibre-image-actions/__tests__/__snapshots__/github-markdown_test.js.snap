// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`writes the markdown with diff report 1`] = `
"Images automagically compressed by [Calibre](https://calibreapp.com)'s [image-actions](https://github.com/marketplace/actions/image-actions) ✨

Compression reduced images by <strong>62.3%</strong>, saving <strong>5.42 KB</strong>.

| Filename | Before | After | Improvement | Visual comparison |
| -------- | ------ | ----- | ----------- | ----------------- |
| <code>icon.png</code> | 8.71 KB | 3.28 KB | -62.3% | [View diff](/calibreapp/image-actions-test-repo/pull/3/commits/a033d6f26da7f7856c150e7f1bf217f0f0cfd7e3?short_path=cc35d4b#diff-cc35d4b8e796c869289176c383aa3da3) |
| <code>roo.jpg</code> | 474.36 KB | 457.91 KB | -3.6% | [View diff](/calibreapp/image-actions-test-repo/pull/3/commits/a033d6f26da7f7856c150e7f1bf217f0f0cfd7e3?short_path=2263610#diff-2263610ef44cc9ef377c71fce4d01b98) |


2 images did not require optimisation.
"
`;

exports[`writes the markdown without diff report 1`] = `
"Images automagically compressed by [Calibre](https://calibreapp.com)'s [image-actions](https://github.com/marketplace/actions/image-actions) ✨

Compression reduced images by <strong>62.3%</strong>, saving <strong>5.42 KB</strong>.

| Filename | Before | After | Improvement |
| -------- | ------ | ----- | ----------- |
| <code>icon.png</code> | 8.71 KB | 3.28 KB | -62.3% |
| <code>roo.jpg</code> | 474.36 KB | 457.91 KB | -3.6% |


2 images did not require optimisation.

"
`;
